torch/ao/quantization/quantizer/x86_inductor_quantizer.py

```diff
@@ -326,7 +442,42 @@ def get_supported_operator_for_quantization_config(
                 return ops
         return []
 
+    def _need_skip_config(
+        self, quantization_config: Optional[QuantizationConfig]
+    ) -> bool:
+        """Check if the provided quantization config is valid for X86InductorQuantizer.
+
+        Mixed static/dynamic configurations or mixed QAT/non-QAT configurations are not supported.
+        If such a mix is detected, the configuration will be marked for skipping..
+        """
+        need_skip = False
```

Comment:
I think we can get current status of static, dynamic, or qat config combining the quantizer's `global_config`, `operator_type_qconfig` and `module_name_qconfig`. Then we can compare the status with the input of `quantization_config` to check whether the input `quantization_config` is valid or not. In this way, we may not need these 2 attributes of `self._is_dynamic` and `self._is_qat`

Reply:
Nice catch, removed the `_is_dynamic` and `_is_qat`, and instead, use `_get_current_quantization_mode` to obtain these status.